src/pip/_internal/index.py

```diff
@@ -694,18 +711,23 @@ def find_candidates(
         self,
         project_name,   # type: str
         specifier=specifiers.SpecifierSet(),  # type: specifiers.BaseSpecifier
+        hashes=None,    # type: Optional[Hashes]
     ):
         """Find matches for the given project and specifier.
 
         If given, `specifier` should implement `filter` to allow version
         filtering (e.g. ``packaging.specifiers.SpecifierSet``).
 
+        If `hashes` is not None, it should be a hash comparer representing
```

Comment:
I'm not sure what the `hashes` stuff is about, but I think it would be better to leave it out of this PR so this PR can just be about clean-up.

Reply:
Sorry, I accidentally base this on another branch I was working on 😞

Reply:
Unpushed